src/Carousel/widget/components/CarouselItem.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * Original code can be found at:
+ * https://github.com/react-bootstrap/react-bootstrap/blob/master/src/CarouselItem.js
+ * 
+ * This code is under the MIT license found here:
+ * https://github.com/react-bootstrap/react-bootstrap/blob/master/LICENSE
+ */
+import classNames = require("Carousel/lib/classnames");
+import * as React from "Carousel/lib/react";
+import ReactDOM = require("Carousel/lib/react-dom");
+
+import { IObject } from "../utils/bootstrapUtils";
+import { Direction } from "./Carousel";
+
+export interface CarouselItemProps extends React.Props<CarouselItem> {
+    active?: boolean;
+    animateIn?: boolean;
+    animateOut?: boolean;
+    direction?: Direction;
+    onAnimateOutEnd?: Function;
+    index?: number;
+    className?: string;
+    onClick?: React.EventHandler<React.MouseEvent<HTMLElement>>;
+    slide?: boolean;
+}
+
+interface CarouselItemState {
+    direction: ItemDirection;
+}
+
+type ItemDirection = "right" | "left";
+
+export class CarouselItem extends React.Component<CarouselItemProps, CarouselItemState> {
+    public static defaultProps: CarouselItemProps = {
+        active: false,
+        animateIn: false,
+        animateOut: false,
+        slide: true,
+    };
+    private isUnmounted: boolean;
+    constructor(props: CarouselItemProps, context: CarouselItem) {
+        super(props, context);
+
+        this.handleAnimateOutEnd = this.handleAnimateOutEnd.bind(this);
+
```

Comment:
These 3 assignments belong together as they all set properties. Keep them together, so remove this white line and the one on line 49.